src/Controller/CacheController.php

```diff
@@ -258,27 +280,45 @@ protected function getCacheName()
         array_unshift($options, null);
         unset($options[0]);
 
-        $answer  = ConsoleSelect::prompt(
-            'You have multiple caches defined, please select one',
-            $options
-        );
+        $this->getConsole()->writeLine('You have multiple caches defined, please select one');
+        
+        foreach ($options as $index => $name) {
+            $this->getConsole()->writeLine(sprintf('%d) %s', $index, $name));
+        }
+
+        do {
+            $answer = (int)ConsoleNumber::prompt(
+                'Please enter a selection: ',
+                false,
+                false,
+                1,
+                count($options)
+            );
+
+            if (!array_key_exists($answer, $options)) {
+                $this->getConsole()->writeLine('Invalid selection', Color::WHITE, Color::RED);
+            }
+        } while (!array_key_exists($answer, $options));
 
         return $options[$answer];
     }
 
     protected function getCaches()
     {
-        $config  = $this->getServiceLocator()->get('Config');
+        $config  = $this->container->get('config');
```

Comment:
@snapshotpl This PR is not a completely refactor, we could do it later in a separate PR, after that this PR will be merged.